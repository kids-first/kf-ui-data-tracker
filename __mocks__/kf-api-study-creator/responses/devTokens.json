{
  "data": {
    "allDevTokens": {
      "edges": [
        {
          "node": {
            "id": "token1",
            "name": "my token 1",
            "createdAt": "2019-06-13T12:05:39.381525+00:00",
            "token": "blahtestingtokenstringhere",
            "creator": {
              "displayName": "John",
              "username": "John",
              "picture": "https://docs.atlassian.com/aui/8.3.1/docs/images/avatar-person.svg",
              "__typename": "UserNode"
            },
            "__typename": "DevDownloadTokenNode"
          },
          "__typename": "DevDownloadTokenNodeEdge"
        },
        {
          "node": {
            "id": "token2",
            "name": "my token 2",
            "createdAt": "2019-06-13T12:05:39.381525+00:00",
            "token": "test***********************",
            "creator": {
              "displayName": "John",
              "username": "John",
              "picture": "https://docs.atlassian.com/aui/8.3.1/docs/images/avatar-person.svg",
              "__typename": "UserNode"
            },
            "__typename": "DevDownloadTokenNode"
          },
          "__typename": "DevDownloadTokenNodeEdge"
        },
        {
          "node": {
            "id": "token3",
            "name": "my token 3",
            "createdAt": "2019-06-13T12:05:39.381525+00:00",
            "token": "test***********************",
            "creator": null,
            "__typename": "DevDownloadTokenNode"
          },
          "__typename": "DevDownloadTokenNodeEdge"
        }
      ],
      "__typename": "DevDownloadTokenNodeConnection"
    }
  }
}
