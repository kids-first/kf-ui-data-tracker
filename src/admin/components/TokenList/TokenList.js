import React from 'react';
import PropTypes from 'prop-types';
import {Button, Popup, Image, Input, List, Icon} from 'semantic-ui-react';
import TimeAgo from 'react-timeago';
import {CopyToClipboard} from 'react-copy-to-clipboard';
import defaultAvatar from '../../../assets/defaultAvatar.png';
import {longDate} from '../../../common/dateUtils';
/**
 * Token list is used to display tokens with the option of copy and delete
 */
const TokenList = ({tokens, deleteToken, newToken}) => (
  <List relaxed>
    {tokens &&
      tokens.length > 0 &&
      tokens.map(node => (
        <List.Item key={node.node.id}>
          <Image
            avatar
            src={
              node.node.creator && node.node.creator.picture
                ? node.node.creator.picture
                : defaultAvatar
            }
            alt={
              node.node.creator && node.node.creator.username
                ? node.node.creator.username
                : 'unknown'
            }
          />
          <List.Content>
            <List.Header>{node.node.name}</List.Header>
            {node.node.creator && <>Created by {node.node.creator.username} </>}
            <TimeAgo
              live={false}
              date={node.node.createdAt}
              title={longDate(node.node.createdAt)}
            />
          </List.Content>
          <List.Content floated="right">
            {newToken && node.node.id === newToken.id ? (
              <Input readOnly action value={newToken.token}>
                <input />
                <Popup
                  trigger={
                    <CopyToClipboard text={newToken.token}>
                      <Button icon>
                        <Icon name="copy" />
                      </Button>
                    </CopyToClipboard>
                  }
                  content="Copied!"
                  on="click"
                />
                <Button
                  icon
                  negative
                  onClick={() => deleteToken(node.node.name)}
                  data-testid="delete-token-button"
                >
                  <Icon name="trash" />
                </Button>
              </Input>
            ) : (
              <Input readOnly action value={node.node.token}>
                <input />
                <Button
                  icon
                  negative
                  onClick={() => deleteToken(node.node.name)}
                  data-testid="delete-token-button"
                >
                  <Icon name="trash" />
                </Button>
              </Input>
            )}
          </List.Content>
        </List.Item>
      ))}
  </List>
);

TokenList.propTypes = {
  /** Array of token object*/
  tokens: PropTypes.array,
  /** Action to delete token*/
  deleteToken: PropTypes.func,
  /** Newly created token when on the page*/
  newToken: PropTypes.object,
};

TokenList.defaultProps = {
  tokens: [],
};

export default TokenList;
