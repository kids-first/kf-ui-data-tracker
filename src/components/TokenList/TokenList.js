import React from 'react';
import {Button, Popup, Image, Input, List, Icon} from 'semantic-ui-react';
import TimeAgo from 'react-timeago';
import {CopyToClipboard} from 'react-copy-to-clipboard';
import defaultAvatar from '../../assets/defaultAvatar.png';

const TokenList = ({tokens, deleteToken}) => (
  <List relaxed>
    {tokens.map(node => (
      <List.Item key={node.node.id}>
        <Image
          avatar
          src={
            node.node.creator && node.node.creator.picture
              ? node.node.creator.picture
              : defaultAvatar
          }
          alt={
            node.node.creator && node.node.creator.username
              ? node.node.creator.username
              : 'unknown'
          }
        />
        <List.Content>
          <List.Header>{node.node.name}</List.Header>
          {node.node.creator && <>Created by {node.node.creator.username} </>}
          <TimeAgo live={false} date={node.node.createdAt} />
        </List.Content>
        <List.Content floated="right">
          <Input readOnly action value={node.node.token}>
            <input />
            {node.node.token.split('*').length !== 24 && (
              <Popup
                trigger={
                  <CopyToClipboard text={node.node.token}>
                    <Button icon>
                      <Icon name="copy" />
                    </Button>
                  </CopyToClipboard>
                }
                content="Copied!"
                on="click"
              />
            )}
            <Button icon negative onClick={() => deleteToken(node.node.name)}>
              <Icon name="trash" />
            </Button>
          </Input>
        </List.Content>
      </List.Item>
    ))}
  </List>
);

export default TokenList;
