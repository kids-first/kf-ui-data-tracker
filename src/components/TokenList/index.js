export {default as TokenList} from './TokenList';
