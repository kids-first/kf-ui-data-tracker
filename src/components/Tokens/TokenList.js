import React from 'react';
import {Button, Container, Header, Image, List, Icon} from 'semantic-ui-react';
import TimeAgo from 'react-timeago';
import CopyButton from '../CopyButton/CopyButton';

const TokenList = ({tokens, deleteToken}) => (
  <List relaxed>
    {tokens.map(node => (
      <List.Item key={node.node.id}>
        <List.Content floated="right">
          <Button icon negative onClick={() => deleteToken(node.node.name)}>
            <Icon name="trash" />
          </Button>
        </List.Content>
        <List.Content floated="right">
          <Header size="tiny" block>
            {node.node.token} <CopyButton text={node.node.token} />
          </Header>
        </List.Content>
        {node.node.creator && <Image avatar src={node.node.creator.picture} />}
        <List.Content>
          <List.Header>{node.node.name}</List.Header>
          {node.node.creator && <>Created by {node.node.creator.username} </>}
          <TimeAgo live={false} date={node.node.createdAt} />
        </List.Content>
      </List.Item>
    ))}
  </List>
);

export default TokenList;
