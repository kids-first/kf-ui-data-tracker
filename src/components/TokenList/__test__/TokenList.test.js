import React from 'react';
import wait from 'waait';
import {render, fireEvent, act} from 'react-testing-library';
import TokenList from '../TokenList';
import devTokens from '../../../../__mocks__/kf-api-study-creator/responses/devTokens.json';

it('renders token list correctly - no data', () => {
  const tree = render(<TokenList />);
  expect(tree.container).toMatchSnapshot();
});

it('renders token list correctly - with new token', async () => {
  const tokens = devTokens.data.allDevTokens.edges;
  const newToken = devTokens.data.allDevTokens.edges[0].node;
  const tree = render(
    <TokenList tokens={tokens} newToken={newToken} deleteToken={jest.fn()} />,
  );

  await wait();
  expect(tree.container).toMatchSnapshot();

  // Click on the delete token button
  act(() => {
    fireEvent.click(tree.getAllByTestId('delete-token-button')[0]);
  });
});
