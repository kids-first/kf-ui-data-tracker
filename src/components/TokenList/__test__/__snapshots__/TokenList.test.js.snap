// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders token list correctly - no data 1`] = `
<div>
  <div
    class="ui relaxed list"
    role="list"
  />
</div>
`;
